sw/host/provisioning/orchestrator/src/device_id.py

```diff
@@ -120,9 +121,14 @@ def __init__(self, sku_config: SkuConfig, din: DeviceIdentificationNumber):
         # Build full device ID.
         self.device_id = (self._sku_specific << 128) | self._base_uid
 
+    def from_hexstr(self, hexstr: str):
+        """Creates a DeviceId object from a hex string."""
+        # TODO: Update other device ID fields based on new device ID.
+        self.device_id = util.parse_hexstring_to_int(hexstr)
```

Comment:
Might be nice to call self.validate() here to ensure the hexstr is valid. Wdyt?

Reply:
Done. This should be taken care of by the DeviceId post_init.